build.gradle

```diff
@@ -0,0 +1,245 @@
+    grpc_netty: "io.grpc:grpc-netty:$grpc_version",
+    grpc_stub: "io.grpc:grpc-stub:$grpc_version",
+    guava: "com.google.guava:guava:$guava_version",
+    guava_testlib: "com.google.guava:guava-testlib:$guava_version",
+    hadoop_client: "org.apache.hadoop:hadoop-client:$hadoop_version",
+    hadoop_common: "org.apache.hadoop:hadoop-common:$hadoop_version",
+    hadoop_mapreduce_client_core: "org.apache.hadoop:hadoop-mapreduce-client-core:$hadoop_version",
+    hadoop_minicluster: "org.apache.hadoop:hadoop-minicluster:$hadoop_version",
+    hadoop_hdfs: "org.apache.hadoop:hadoop-hdfs:$hadoop_version",
+    hadoop_hdfs_tests: "org.apache.hadoop:hadoop-hdfs:$hadoop_version:tests",
+    hamcrest_core: "org.hamcrest:hamcrest-core:$hamcrest_version",
+    hamcrest_library: "org.hamcrest:hamcrest-library:$hamcrest_version",
+    jackson_annotations: "com.fasterxml.jackson.core:jackson-annotations:$jackson_version",
+    jackson_core: "com.fasterxml.jackson.core:jackson-core:$jackson_version",
+    jackson_databind: "com.fasterxml.jackson.core:jackson-databind:$jackson_version",
+    jackson_dataformat_yaml: "com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:$jackson_version",
+    jackson_module_scala: "com.fasterxml.jackson.module:jackson-module-scala_2.10:$jackson_version",
+    joda_time: "joda-time:joda-time:2.4",
+    junit: "junit:junit:4.12",
+    kafka_clients: "org.apache.kafka:kafka-clients:0.10.1.0",
+    malhar_library: "org.apache.apex:malhar-library:$apex_malhar_version",
+    mockito_core: "org.mockito:mockito-core:1.9.5",
+    netty_handler: "io.netty:netty-handler:$netty_version",
+    netty_tcnative_boringssl_static: "io.netty:netty-tcnative-boringssl-static:1.1.33.Fork26",
+    netty_transport_native_epoll: "io.netty:netty-transport-native-epoll:$netty_version",
+    protobuf_java: "com.google.protobuf:protobuf-java:$protobuf_version",
+    protobuf_java_util: "com.google.protobuf:protobuf-java-util:$protobuf_version",
+    proto_google_cloud_pubsub_v1: "com.google.api.grpc:proto-google-cloud-pubsub-v1:$pubsub_grpc_version",
+    proto_google_cloud_spanner_admin_database_v1: "com.google.api.grpc:proto-google-cloud-spanner-admin-database-v1:$grpc_google_common_protos",
+    proto_google_common_protos: "com.google.api.grpc:proto-google-common-protos:$grpc_google_common_protos",
+    slf4j_api: "org.slf4j:slf4j-api:1.7.25",
+    slf4j_jdk14: "org.slf4j:slf4j-jdk14:1.7.25",
+    slf4j_log4j12: "org.slf4j:slf4j-log4j12:1.7.25",
+    snappy_java: "org.xerial.snappy:snappy-java:1.1.4",
+    spark_core: "org.apache.spark:spark-core_2.10:$spark_version",
+    spark_network_common: "org.apache.spark:spark-network-common_2.10:$spark_version",
+    spark_streaming: "org.apache.spark:spark-streaming_2.10:$spark_version",
+    stax2_api: "org.codehaus.woodstox:stax2-api:3.1.4",
+    woodstox_core_asl: "org.codehaus.woodstox:woodstox-core-asl:4.4.1",
+  ],
+]
+
+buildscript {
```

Comment:
<!--new_thread; commit:5886e58537a4f58de6611a40e5e8e09655913ae0; resolved:0-->
Since many people will not have used Gradle, and even those that have will edit builds less often than their main code, can you signpost the significance of sections and commands? Especially commenting on any that affect whole-project things, like this maven config looks like it does.

Reply:
I commented build_rules.gradle heavily for parts that I had a lot of knowledge in and gave some thought to what made sense.

Also some parts of build.gradle.